src/main/java/com/example/bookshop/exception/CustomGlobalExceptionHandler.java

```diff
@@ -0,0 +1,47 @@
+package com.example.bookshop.exception;
+
+import java.time.LocalDateTime;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+
+@ControllerAdvice
+public class CustomGlobalExceptionHandler extends ResponseEntityExceptionHandler {
+    @ExceptionHandler({DataProcessingException.class, EntityNotFoundException.class})

```

Comment:
this is wrong


this method handles `MethodArgumentNotValidException` only

for other exceptions such as `DataProcessingException`, `EntityNotFoundException` you should create separate `@ExceptionHandler` methods:
```
@ExceptionHandler(EntityNotFoundException.class)
public ResponseEntity<String> handleEntityNotFound(EntityNotFoundException ex) {
  return ResponseEntity.status(HttpStatus.NOT_FOUND)
                   .body(ex.getMessage());
}
```